nilearn/experimental/surface/_surface_image.py

```diff
@@ -0,0 +1,84 @@
+from __future__ import annotations
+
+import dataclasses
+import pathlib
+from typing import Dict
+
+import numpy as np
+
+from nilearn.experimental.surface import _io
+
+PolyData = Dict[str, np.ndarray]
+
+
+class Mesh:
+    n_vertices: int
+
+    _file: pathlib.Path
+
+    def __init__(self, file_path: pathlib.Path | str) -> None:
+        self._file = pathlib.Path(file_path)
+        self.n_vertices = _io.read_mesh(self._file)["coordinates"].shape[0]
+
+    def __repr__(self) -> str:
+        return f"<{self.__class__.__name__} with {self.n_vertices} vertices>"
+
+    # TODO those are properties for now for compatibility with plot_surf_img
+    # for the demo but should become functions as they can take some time to
+    # return
+    @property
+    def coordinates(self) -> np.ndarray:
+        return _io.read_mesh(self._file)["coordinates"]
+
+    @property
+    def faces(self) -> np.ndarray:
+        return _io.read_mesh(self._file)["faces"]
+
+
+class InMemoryMesh(Mesh):
```

Comment:
Why not just expand the type of input for class `Mesh` since `load_surf_mesh` that's called in `read_mesh` takes in loaded arrays

Reply:
for now it uses load_surf_mesh just so we can focus on the surface image and maskers first and deal with the io later. but I think it might simplify things to have a distinction between files on disk and in-memory meshes, to avoid (i) functions like `load_surf_mesh` that have parameters that can be of several completely different types and (ii) a surface image that has several redundant/alternative attributes, some of which can be missing, and always having to deal with those multiple possible configurations